src/main/kotlin/ecommerce/model/Product.kt

```
@@ -0,0 +1,41 @@
+package ecommerce.model
+
+import jakarta.persistence.CascadeType
+import jakarta.persistence.Column
+import jakarta.persistence.Entity
+import jakarta.persistence.FetchType
+import jakarta.persistence.GeneratedValue
+import jakarta.persistence.GenerationType
+import jakarta.persistence.Id
+import jakarta.persistence.OneToMany
+import jakarta.persistence.Table
+
+@Entity
+@Table(name = "products")
+class Product(
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    var id: Long? = null,
```

I see you've used `Long?` type for id 👍 
There are some other people who used `Long` and gave `0L` as the value for 'non existing' id.
What do you think the differences are?
(To give you some more information, I also prefer `null` so you are not wrong 👍 )

---

I have used a nullable type, because DB indexes could also start at 0, so null makes it more clear that the entity either does not yet have an index or the index was not retrieved from the DB. This way, my entity can also act as a model and I don't have to pass around the ID every time.

---

Of course, we have to use safe-call operators, elvis operators or not-null assertion operators to avoid NullPointerException. That made me consider switching to `0L`, but null was oftentimes more convenient for our purposes.

---

Great!
I prefer `null` over `0L` because it keeps the meaning of “non-existing” consistent across all types in the project.
For some classes, defining a default placeholder value is unclear or misleading, so `null` is the clearest option. While using `null` does introduce the possibility of NPEs, Kotlin’s null-safety features make them manageable and easy to detect at compile time.